named_arrays/_matrices/matrices.py

```diff
@@ -33,8 +34,17 @@ def components(self) -> dict[str, na.AbstractVectorArray]:
 
     @property
     def entries(self) -> dict[tuple[str, str], na.ScalarLike]:
-        rows = self.rows
-        return {(r, c): rows[r].components[c] for r in rows for c in rows[r].components}
+        rows = self.cartesian_nd.rows
+        result = {}
+        for r in rows:
+            row = rows[r]
+            if isinstance(row, na.AbstractArray):
+                for c in row.components:
+                    result[(r, c)] = row.components[c]
+            else:
+                result[(r, '')] = row
```

Comment:
I think this would make more sense to me as a 1-element tuple. Would that break anything if we change it to that?